ambry-rest/src/main/java/com.github.ambry.rest/NettyMessageProcessor.java

```diff
@@ -203,35 +215,40 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object event) {
   @Override
   public void channelRead0(ChannelHandlerContext ctx, HttpObject obj)
       throws RestServiceException {
-    logger.trace("Reading on channel {}", ctx.channel());
-    long currentTime = System.currentTimeMillis();
+    if (isOpen()) {
+      logger.trace("Reading on channel {}", ctx.channel());
+      long currentTime = System.currentTimeMillis();
 
-    boolean recognized = false;
-    boolean success = true;
-    if (obj instanceof HttpRequest) {
-      recognized = true;
-      success = handleRequest((HttpRequest) obj);
-    }
-    // this is an if and not an else-if because a HttpObject can be both HttpRequest and HttpContent.
-    if (success && obj instanceof HttpContent) {
-      recognized = true;
-      success = handleContent((HttpContent) obj);
-    }
-    if (success && !recognized) {
-      logger.warn("Received null/unrecognized HttpObject {} on channel {}", obj, ctx.channel());
-      nettyMetrics.unknownHttpObjectError.inc();
-      if (responseChannel == null || requestContentFullyReceived) {
-        resetState();
+      boolean recognized = false;
+      boolean success = true;
+      if (obj instanceof HttpRequest) {
+        recognized = true;
+        success = handleRequest((HttpRequest) obj);
+      }
+      // this is an if and not an else-if because a HttpObject can be both HttpRequest and HttpContent.
+      if (success && obj instanceof HttpContent) {
+        recognized = true;
+        success = handleContent((HttpContent) obj);
+      }
+      if (success && !recognized) {
+        logger.warn("Received null/unrecognized HttpObject {} on channel {}", obj, ctx.channel());
+        nettyMetrics.unknownHttpObjectError.inc();
+        if (responseChannel == null || requestContentFullyReceived) {
+          resetState();
+        }
+        onRequestAborted(new RestServiceException("HttpObject received is null or not of a known type",
+            RestServiceErrorCode.UnknownHttpObject));
       }
-      responseChannel.onResponseComplete(new RestServiceException("HttpObject received is null or not of a known type",
```

Comment:
It has not been removed. This is there in line 239 of  (new) `NettyMessageProcessor`.
```
onRequestAborted(new RestServiceException("HttpObject received is null or not of a known type", RestServiceErrorCode.UnknownHttpObject));
```